workorders.lic

```diff
@@ -294,29 +294,53 @@ class WorkOrders
     if stock_needed < quantity
       order_rem(info['catalyst-room'], stock_needed, info['catalyst_number'])
     end
-    case bput("get water from my #{@bag}", 'What were', 'You get')
-    when 'What were'
-      stock_needed = 1
-      order_rem(info['stock-room'], stock_needed, info['stock-number'])
-    when 'You get'
-      /are (.*) parts left of the water|is only (.*) part left of the water/ =~ bput('count my water', 'are .* parts left of the water', 'is only .* part left of the water')
-      water = Regexp.last_match(1) ? Regexp.last_match(1).scan(/\w+/).to_a : Regexp.last_match(2).scan(/\w+/).to_a
-      val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'twenty-one' => 21, 'twenty-two' => 22, 'twenty-three' => 23, 'twenty-four' => 24, 'twenty-five' => 25, 'twenty-six' => 26, 'twenty-seven' => 27, 'twenty-eight' => 28, 'twenty-nine' => 29, 'thirty' => 30, 'thirty-one' => 31, 'thirty-two' => 32, 'thirty-three' => 33, 'thirty-four' => 34, 'thirty-five' => 35, 'thirty-six' => 36, 'thirty-seven' => 37, 'thirty-eight' => 38, 'thirty-nine' => 39, 'forty' => 40, 'forty-one' => 41, 'forty-two' => 42, 'forty-three' => 43, 'forty-four' => 44, 'forty-five' => 45, 'forty-six' => 46, 'forty-seven' => 47, 'forty-eight' => 48, 'forty-nine' => 49, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
-      value = water.map { |word| val_map[word] }.inject(&:+)
-      stock_needed = ((quantity * recipe['volume'] - value) / 100.0).ceil
-    end
+    if recipe['liquid'].nil?
```

Comment:
I have changed the way this part is handled and testing.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

On 09/11/2016 01:29 PM, Robert Uhl wrote:
>
> In workorders.lic 
> <https://github.com/rpherbig/dr-scripts/pull/776#discussion_r78298951>:
>
> > @@ -294,29 +294,53 @@ class WorkOrders
> >      if stock_needed < quantity
> >        order_rem(info['catalyst-room'], stock_needed, info['catalyst_number'])
> >      end
> > -    case bput("get water from my #{@bag}", 'What were', 'You get')
> > -    when 'What were'
> > -      stock_needed = 1
> > -      order_rem(info['stock-room'], stock_needed, info['stock-number'])
> > -    when 'You get'
> > -      /are (.*) parts left of the water|is only (.*) part left of the water/ =~ bput('count my water', 'are .* parts left of the water', 'is only .* part left of the water')
> > -      water = Regexp.last_match(1) ? Regexp.last_match(1).scan(/\w+/).to_a : Regexp.last_match(2).scan(/\w+/).to_a
> > -      val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'twenty-one' => 21, 'twenty-two' => 22, 'twenty-three' => 23, 'twenty-four' => 24, 'twenty-five' => 25, 'twenty-six' => 26, 'twenty-seven' => 27, 'twenty-eight' => 28, 'twenty-nine' => 29, 'thirty' => 30, 'thirty-one' => 31, 'thirty-two' => 32, 'thirty-three' => 33, 'thirty-four' => 34, 'thirty-five' => 35, 'thirty-six' => 36, 'thirty-seven' => 37, 'thirty-eight' => 38, 'thirty-nine' => 39, 'forty' => 40, 'forty-one' => 41, 'forty-two' => 42, 'forty-three' => 43, 'forty-four' => 44, 'forty-five' => 45, 'forty-six' => 46, 'forty-seven' => 47, 'forty-eight' => 48, 'forty-nine' => 49, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
> > -      value = water.map { |word| val_map[word] }.inject(&:+)
> > -      stock_needed = ((quantity * recipe['volume'] - value) / 100.0).ceil
> > -    end
> > +    if recipe['liquid'].nil?
>
> Let's add water entries or default the recipe liquid to water instead 
> off doing these two nearly identical actions for water and alcohol.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/rpherbig/dr-scripts/pull/776/files/c1462a6f9b979c1406d8937c25882bc3419f1426#r78298951>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AU8XXPLJNW-jOG3gwjg5ZljrWCFhxuoFks5qpDpggaJpZM4J531I>.
>